libraries/AP_Scripting/generator/description/bindings.desc

```diff
@@ -328,3 +328,15 @@ singleton AP_MotorsMatrix_6DoF_Scripting depends APM_BUILD_TYPE(APM_BUILD_ArduCo
 singleton AP_MotorsMatrix_6DoF_Scripting alias Motors_6DoF
 singleton AP_MotorsMatrix_6DoF_Scripting method init boolean uint8_t 0 (AP_MOTORS_MAX_NUM_MOTORS-1)
 singleton AP_MotorsMatrix_6DoF_Scripting method add_motor void int8_t 0 (AP_MOTORS_MAX_NUM_MOTORS-1) float -FLT_MAX FLT_MAX float -FLT_MAX FLT_MAX float -FLT_MAX FLT_MAX float -FLT_MAX FLT_MAX float -FLT_MAX FLT_MAX float -FLT_MAX FLT_MAX boolean uint8_t 0 (AP_MOTORS_MAX_NUM_MOTORS-1)
+
+include AP_HAL/I2CDevice.h
+ap_object AP_HAL::I2CDevice semaphore-pointer
+ap_object AP_HAL::I2CDevice method set_retries void uint8_t 1 20
+ap_object AP_HAL::I2CDevice method write_register boolean uint8_t 0 UINT8_MAX uint8_t 0 UINT8_MAX
+ap_object AP_HAL::I2CDevice method read_registers boolean uint8_t 0 UINT8_MAX &uint8_t'Null 1'literal
+ap_object AP_HAL::I2CDevice method set_address void uint8_t 0 UINT8_MAX
+
+singleton hal.i2c_mgr alias i2c
+singleton hal.i2c_mgr literal
+singleton hal.i2c_mgr method get_device_direct AP_HAL::I2CDevice uint8_t 0 4 uint8_t 0 UINT8_MAX
```

Comment:
4 should probably be a define we get from the HAL somewhere for the number of buses possible. Also probably worth considering that you almost guranteed need the clock speed, and allowing using SMBus would actually be valuable as a flag.

Reply:
I don't think there is a unified define for this over all the HALs

Reply:
added clock speed and SMBus flag